editor/gui/editor_scene_tabs.cpp

```diff
@@ -190,59 +190,87 @@ void EditorSceneTabs::_disable_menu_option_if(int p_option, bool p_condition) {
 	}
 }
 
-// TODO: This REALLY should be done in a better way than replacing all tabs after almost EVERY action.
 void EditorSceneTabs::update_scene_tabs() {
 	tab_preview_panel->hide();
 
-	bool show_rb = EDITOR_GET("interface/scene_tabs/show_script_button");
+	if (scene_tabs->get_tab_count() == EditorNode::get_editor_data().get_edited_scene_count()) {
+		_update_tab_titles();
+		return;
+	}
 
 	if (DisplayServer::get_singleton()->has_feature(DisplayServer::FEATURE_GLOBAL_MENU)) {
 		DisplayServer::get_singleton()->global_menu_clear("_dock");
 	}
 
+	// Workaround to ignore the tab_changed signal.
+	scene_tabs->disconnect("tab_changed", callable_mp(this, &EditorSceneTabs::_scene_tab_changed));
```

Comment:
Since you are moving this code anyway, it can be changed to use `set_block_signals()` instead of disconnecting.